app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java

```diff
@@ -162,15 +163,44 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
             final String hostname = addressET.getText().toString();
             final int port = Integer.parseInt(portET.getText().toString());
             final String username = usernameET.getText().toString();
-            final String password = passwordET.getText() != null ?
-                    passwordET.getText().toString() : null;
+            final String password = !TextUtils.isEmpty(passwordET.getText()) ?
+                    passwordET.getText().toString() : getArguments().getString("password", null);
```

Comment:
I think we can simplify this logic into:
```Java
isEmpty(...) ? getArguments().getString("password", null) : passwordET.getText().toString();
```

to make things less complicated to maintain.